Added the 1st draft of the Annotation implementation

songpeize opened the pull request.

songpeize committed:
Added basic annotation implementation

- Waiting to test with the headset

songpeize committed:
Annotation Update

- Fade out effect
- Better pointer
- Canvas following head position
- Adding picture in Canvas

songpeize committed:
Fixed issues found when testing with headset

- Fixed the roll rotation when users tilt the head sideways
- Fixed the issues with fading out

songpeize committed:
Added video playback and more canvas control

songpeize committed:
Fixed annotation pointer end position

songpeize committed:
Start working on annotation interface

songpeize committed:
Updated annotation pointer collider

songpeize committed:
Added real-time annotation attachment and cooking object

songpeize committed:
Improved real-time demo by having the program detecting newly added detected object

songpeize committed:
Finished demo with professor

- Added content for the demo
- Capsule collider for the pointer
- Delay for the annotation

songpeize committed:
Push the version before demo

songpeize committed:
Working on the gaze cone functionality

songpeize committed:
Added gaze sphere for object selection

songpeize committed:
Updated Annotation Anchor Visibility and Refined eye gaze selection

songpeize committed:
Updated logic for detecting last hit object for annotation

songpeize committed:
Updated fixation and saccade detection using I-VT algorithm

songpeize committed:
Tested fixation change on headsets, adjusted some parameters based on testing

songpeize committed:
Updated for the latest demo

songpeize committed:
Debugging disabling issue; updated the scene for Demo with Bettina

ovenmitt commented:
Could you put all resources that are not needed for the ARUI, only for testing (e..g, the videos and images) in a separate folder and mark them as temporary? e.g., in the Assets folder create a new folder and call it 'tmp' or 'DEMO'

songpeize committed:
Organized demo medias into folders

songpeize committed:
Bug fix for annotation selection

- Fixed a bug that sometimes the annotation pointer activates but the annotation canvas stays deactivated.
- Fixed a bug that annotation doesn't react even when the user is looking at it.

songpeize committed:
Added demo for dual gaze method

songpeize committed:
Improved DualGaze Demo; Added Eye+Head gaze demo

songpeize committed:
Implemented multi-selection and headGaze selection

- Todo: I-VT algorithm

songpeize committed:
Finished the essential functionalities of the system with a demo scene
